src/reflection_compat.jl

```diff
@@ -102,13 +102,21 @@ function code_warntype_by_type(io::IO, @nospecialize(tt);
                 end
                 print(io, "  ", slotnames[i])
                 if isa(slottypes, Vector{Any})
-                    InteractiveUtils.warntype_type_printer(io, slottypes[i], true)
+                    @static if hasmethod(InteractiveUtils.warntype_type_printer, (IO,))
```

Comment:
I strongly prefer version numbers since these tests can be accidentally true.

Reply:
Ah, right. I switched to `VERSION`-based check.